Sources/TemplateRef.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  TemplateRef.swift
+//  swiftgen
+//
+//  Created by Olivier HALLIGON on 11/10/2017.
+//  Copyright © 2017 AliSoftware. All rights reserved.
+//
+
+import PathKit
+
+enum TemplateRef {
```

Comment:
Nice! (a bit similar to the output destination, I really like this 👍 )

Reply:
Yeah, it was about time we turn that exclusive choice of name vs path into an enum, I wonder why I didn't do it sooner 😄

Reply:
I even wondered if it wouldn't be worth it making it `ArgumentConvertible` like `OutputDestination`. But that would mean merging the `-t` and `-p` into one single option (and interpreting it as a `TemplateRef.path` if it contains a `/` and a `TemplateRef.name` otherwise, for example?). Would be too much breaking change for this, at least for now.

Reply:
It would be too much of a breaking change, and a bit confusing for users. Also, paths might not always include a `/` character, so how would that even work?

Reply:
Yeah I also thought of other options (telling people to use `./foo` to disambiguate between name and local-path-in-same-dir but not ideal. Or on the contrary having to prefix template name with `:` or something to refer to a name instead of a path (eg `-t :swift3`) but that would be even more breaking (and using a name is the most common usage compared to using as path so path should be the exception not the other way around)… in the end none of those scenarios feels ideal 😕 so let's keep it as is for now 😉